framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/AmqDeploymentImpl.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.cloud.openshift.deployment;
+
+import java.net.URL;
+import java.util.Optional;
+
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteBuilder;
+import org.kie.cloud.api.deployment.AmqDeployment;
+import org.kie.cloud.api.deployment.constants.DeploymentConstants;
+import org.kie.cloud.openshift.resource.Project;
+
+public class AmqDeploymentImpl extends OpenShiftDeployment implements AmqDeployment {
+
+    private String serviceName, tcpServiceName;
+    private Optional<URL> insecureUrl;
+    private Optional<URL> secureUrl;
+    private Optional<URL> tcpUrl;
+    private String username;
+    private String password;
+
+    public AmqDeploymentImpl(Project project) {
+        super(project);
+    }
+
+    @Override
+    public String getServiceName() {
+        if (serviceName == null) {
+            serviceName = ServiceUtil.getAmqJolokiaServiceName(getOpenShift());
+        }
+        return serviceName;
+    }
+
+    private String getAmqTcpServiceName() {
+        if (tcpServiceName == null) {
+            tcpServiceName = ServiceUtil.getAmqTcpServiceName(getOpenShift());
+        }
+        return tcpServiceName;
+    }
+
+    private Optional<URL> createSecureTcpUrl() {
+        String amqTcpHost = getAmqTcpServiceName() + "-" + getNamespace()
+                + DeploymentConstants.getDefaultDomainSuffix();
+
+        Route tcpRoute = new RouteBuilder()
+                .withNewMetadata()
+                    .withName(getAmqTcpServiceName())
+                .endMetadata()
+                .withNewSpec()
+                    .withNewTo()
+                        .withKind("Service")
+                        .withName(getAmqTcpServiceName()+"-ssl")
+                    .endTo()
+                    .withNewPort()
+                        .withNewTargetPort("amq-tcp-ssl")
+                    .endPort()
+                    .withNewTls()
+                        .withTermination("passthrough")
+                    .endTls()
+                    .withHost(amqTcpHost)
+                    .withWildcardPolicy("None")
+                .endSpec()
+                .build();
+
+        getOpenShift().routes().createOrReplace(tcpRoute);
+
+        return getHttpsRouteUrl(getAmqTcpServiceName()+"-ssl");
+    }
+
+    @Override
+    public URL getTcpUrl() {
```

Comment:
Why do you handle new Route here, considering there is publicly available TCP SSL route - https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/master/templates/rhpam75-prod-immutable-kieserver-amq.yaml#L809

Reply:
This route was previously not include in the template.